### PR TITLE
Added onPageBeingChanged event

### PR DESCRIPTION
**Pull request by gabrieletondi**
Hi,

Together with @thierryfornetti we have added a new event "onPageBeingChanged" that will be triggered every time the user start the scroll to a new page.

Our use case was: hide some UI elements when a page is appearing.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -94,6 +96,9 @@ export default class Carousel extends Component {
     } else {
       this.state = { size };
     }
+    this.offset = 0;
+    this.direction = 'right';
```
Comment (phil-r) [inline index.js]: This is not really needed

File: index.js [hunk 2]
```diff
@@ -291,6 +296,26 @@ export default class Carousel extends Component {
     return this._normalizePageNumber(page);
   }
 
+  _onScroll = (event) => {
+    const currentOffset = event.nativeEvent.contentOffset.x;
+    this.direction = currentOffset > this.offset ? 'right' : 'left';
```
Comment (phil-r) [inline index.js]: I think it's better to use `const direction = ...` here

File: index.js [hunk 3]
```diff
@@ -291,6 +296,26 @@ export default class Carousel extends Component {
...
+    this.offset = currentOffset;
+    const nextPage = this._calculateNextPage();
```
Comment (phil-r) [inline index.js]: and pass `direction` as an argument here

File: index.js [hunk 4]
```diff
@@ -291,6 +296,26 @@ export default class Carousel extends Component {
...
+    if (this.nextPage !== nextPage) {
+      this.nextPage = nextPage;
+      if (this.props.onPageBeingChanged) {
+        this.props.onPageBeingChanged(this.nextPage);
+      }
+    }
+  }
+
+  _calculateNextPage = () => {
```
Comment (phil-r) [inline index.js]: and take `direction` here

**Comment by gabrieletondi** [comment]
Hi @phil-r!
Thank you for the review!

I've just pushed the changed required.

Thank you for your time!
Gabriele

**Pull request merged**